File: src/apps/Andriod/app/src/main/java/com/dkellycollins/triggerfinger/managers/entity/ICollidableEntityManager.java
package com.dkellycollins.triggerfinger.managers.entity;

import com.dkellycollins.triggerfinger.data.entity.ICollidable;
import com.dkellycollins.triggerfinger.data.model.IVector;

public interface ICollidableEntityManager extends IEntityManager {

    Iterable<ICollidable> retrieve();

    ICollidable retrieve(int id);

    int create(IVector position, float radius);

    void update(int id, IVector position, float radius);

    void delete(int id);

}

File: src/apps/Android-Kotlin/app/src/main/kotlin/com/dkellycollins/triggerfinger/managers/entity/ICollidableEntityManager.kt
package com.dkellycollins.triggerfinger.managers.entity

import com.dkellycollins.triggerfinger.data.entity.ICollidable
import com.dkellycollins.triggerfinger.data.model.IVector

interface ICollidableEntityManager : IEntityManager {

    fun retrieve(): Iterable<ICollidable>

    fun retrieve(id: Int): ICollidable

    fun create(position: IVector, radius: Float): Int

    fun update(id: Int, position: IVector, radius: Float)

    fun delete(id: Int)

}
